scripts/scripts.js

```diff
@@ -56,6 +58,168 @@ function buildAutoBlocks(main) {
   }
 }
 
+/**
+ * Click event handler for social share anchors.
+ * Opens the social share dialog having social buttons.
+ *
+ * @param {Element} anchor - The anchor element clicked.
+ */
+export async function handleSocialShareClick(anchor) {
+  const dialogs = document.querySelectorAll('dialog.social-share');
+  const placeholders = await fetchPlaceholders();
+  const link = anchor.closest('li')?.querySelector('a').href ?? anchor.dataset?.href;
```

Comment:
I am trying to use this to create social share dialog on videos-podcasts block. IMO to make it more independent from the structure of the block, shouldn't we pass the link which need to be share instead of searching for closet `li` and anchor inside to get the link.
As in my case the link is in `h3` not in `li` so it will not the link which need to shared.
thoughts ?

Reply:
BTW, sending direct link in the method it self well worked for me.
`export async function handleSocialShareClick(shareLink) {...}`

Reply:
Make scene, we need to cover all cases.

Reply:
It has fallback for anchor.href when closes `li` is not found. I hope it is working for you

Reply:
yes it worked but with a twist,
If I send an anchor which have href then on clicking on that anchor it open share dialog but redirect as well immediately , so i have used it like.
`socialAnchor.addEventListener('click', (event) => {
    event.preventDefault();
    handleSocialShareClick(socialAnchor);
  });`

Reply:
Adapted to use `handleSocialShareClick` as it provided in the PR

Reply:
you don't need that twist, just check if the following code is getting called:
https://github.com/aemsites/icicidirect/pull/82/files#diff-fbceb58f32b99b8a5309251f8c49ac8eda82fba58b905c600c92bce1ed2a1c8cR180-R183

Or you can set data-href on anchor element, instead of href.